code/modules/power/solar.dm

```diff
@@ -98,80 +108,94 @@
 	qdel(src)
 
 
-/obj/machinery/power/solar/update_icon()
-	..()
-	cut_overlays()
+/obj/machinery/power/solar/update_overlays()
+	. = ..()
+	var/matrix/turner = matrix()
+	turner.Turn(azimuth_current)
+	panel.transform = turner
 	if(machine_stat & BROKEN)
-		add_overlay(mutable_appearance(icon, "solar_panel-b", FLY_LAYER))
+		panel.icon_state = "solar_panel-b"
 	else
-		add_overlay(mutable_appearance(icon, "solar_panel", FLY_LAYER))
-		src.setDir(angle2dir(adir))
+		panel.icon_state = "solar_panel"
+#if DM_VERSION <= 512
```

Comment:
You already have the better code for solar panels implemented, just remove the >= 513 stuff, since mutable appearances are better than vis contents.